substrate/frame/staking/src/lib.rs

```diff
@@ -1127,10 +1131,64 @@ impl<T: Config> Convert<T::AccountId, Option<Exposure<T::AccountId, BalanceOf<T>
 	}
 }
 
-pub struct NullIdentity;
-impl<T> Convert<T, Option<()>> for NullIdentity {
-	fn convert(_: T) -> Option<()> {
-		Some(())
+/// A marker type representing the presence of a validator. Encodes as a unit type.
+pub type Existence = ();
+
+/// A converter type that returns `Some(())` if the validator exists in the current active era,
+/// otherwise `None`. This serves as a lightweight presence check for validators.
+pub struct ExistenceOf<T>(core::marker::PhantomData<T>);
```

Comment:
We should move this to `stkaing-next` branch, right? 

`staking-classic` only needs `ExistenceOrLegacyExposureOf`.

Reply:
We need them only in staking-classic, right? `Staking-next` does not interact with historical sessions atleast directly?

We don't strictly need `ExistenceOf`. But seemed a nice thing to have. New solo chains can choose `ExistenceOf` instead of `ExistenceOrLegacyExposureOf `.